components/Tabs/Tabs.js

```diff
@@ -1,56 +1,48 @@
 class TabLink {
-  constructor(element){
-    // assign this.element to the element reference
-    this.element;
-    // Get the tab data attribute and save the value here
-    this.tabData = ; 
-    // Find all elements with the .card class in index.html that correspond to the tab data attribute. If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+  constructor(element) {
+    this.element = element;
+    // get tab data set values
+    this.tabData = this.element.dataset.tab;
+
+    if (this.tabData === "all") {
+      // if all tabs is selected, grab all cards with .card class
+      this.cards = document.querySelectorAll('.card');
     } else {
-      this.cards = ;
+      // if any other card is selected, grab individual dataset
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
```

Comment:
Great job in interpreting the objectives asked of you into hard, functional code.